src/VisualStudio/Core/Def/ProjectSystem/FileChangeWatcher.cs

```diff
@@ -367,8 +361,13 @@ public Context(FileChangeWatcher fileChangeWatcher, ImmutableArray<WatchedDirect
 
             foreach (var watchedDirectory in watchedDirectories)
             {
-                _fileChangeWatcher._taskQueue.AddWork(watchedDirectories.Select(
-                    watchedDirectory => WatcherOperation.WatchDirectory(watchedDirectory.Path, watchedDirectory.ExtensionFilter, this, _directoryWatchCookies)));
+                var item = WatcherOperation.WatchDirectory(
```

Comment:
>var item = WatcherOperation.WatchDirectory( [](http://example.com/codeflow?start=16&length=43)

Noticed there was a bug here in that every watched directory caused an operation to be created for the whole immutable array, not just for itself.

Reply:
Good catch!